src/mds/journal.cc

```diff
@@ -1726,6 +1726,7 @@ void ESession::replay(MDSRank *mds)
       session = mds->sessionmap.get_or_add_session(client_inst);
       mds->sessionmap.set_state(session, Session::STATE_OPEN);
       session->set_client_metadata(client_metadata);
+      session->set_client_not_open(false);
```

Comment:
no code submits the ESession event

Reply:
Here It's just clearing the client_not_open flag set during EImportStart replay. Meaning the client had actually opened the connection and so we don't need this flag to be set. So during reconnect if the flag is set and the client failed to reconnect, then we kill the session and if it is not set and client failed to reconnect, then blacklist.

Reply:
check Server::handle_client_session(). if session is already being opened by Server::prepare_force_open_sessions. it does not submit ESession event